nextcord/client.py

```diff
@@ -2096,33 +2111,170 @@ def get_application_command(self, command_id: int) -> Optional[BaseApplicationCo
 
+        .. versionadded:: 2.0
+
+        .. versionchanged:: 2.4
+            it's no longer possible to pass ``None`` to the ``name`` parameter.
+
+        .. versionchanged:: 2.4
+            Renamed the ``name`` parameter to ``qualified_name``.
+
+        .. versionchanged:: 2.4
+            Renamed the ``cmd_type`` parameter to ``type``.
+
+        .. versionchanged:: 2.4
+            Added the ``search_locales`` keyword argument.
+
+        .. versionchanged:: 2.4
+            Subcommands/Subcommand groups can now be retrieved with this method.
+
+        .. versionchanged:: 2.4
+            Changed the signature from ``(qualified_name, type, guild_id, *, search_locales)`` to ``(*, type, qualified_name, guild_id, search_locales)``.
+            All parameters are now keyword-only.
+
         Parameters
         ----------
-        name: :class:`str`
-            Name of the application command. Capital sensitive.
-        cmd_type: Union[:class:`int`, :class:`ApplicationCommandType`]
+        type: Union[:class:`int`, :class:`ApplicationCommandType`]
             Type of application command.
+        qualified_name: :class:`str`
+            Full name of the application command. Case sensitive.
+            Subcommands must be separated by a space, E.g, ``parent group subcommand``.
         guild_id: Optional[:class:`int`]
             Guild ID of the signature. If set to ``None``, it will attempt to get the global signature.
+        search_locales: :class:`bool`
+            Whether to also search through the command's name locales. Defaults to ``False``.
 
         Returns
         -------
-        command: Optional[:class:`BaseApplicationCommand`]
-            Application Command with the given signature. If no command with that signature is
-            found, returns ``None`` instead.
+        command: Optional[:class:`BaseApplicationCommand`, :class:`SlashApplicationSubcommand`]
+            Application Command with the given signature. If no command with that signature is found, ``None`` is returned instead.
         """
-        if isinstance(cmd_type, ApplicationCommandType):
-            actual_type = cmd_type.value
-        else:
-            actual_type = cmd_type
-
         return self._connection.get_application_command_from_signature(
-            name=name, cmd_type=actual_type, guild_id=guild_id
+            type=type.value if isinstance(type, ApplicationCommandType) else type,
+            qualified_name=qualified_name,
+            guild_id=guild_id,
+            search_locales=search_locales,
+        )
+
+    @overload
+    def get_application_command_from_signature(
+        self,
+        *,
+        type: Union[int, ApplicationCommandType],
+        qualified_name: str,
+        guild_id: Optional[int],
+        search_locales: bool = ...,
+    ) -> Optional[Union[BaseApplicationCommand, SlashApplicationSubcommand]]:
+        ...
+
+    @overload
+    def get_application_command_from_signature(
```

Comment:
Shouldn't there also be overloads where the parameters have their old names? The old signature isn't positional only, after all.